mdbenchmark/utils.py

```diff
@@ -125,3 +128,24 @@ def guess_ncores():
         )
         total_cores = 1
     return total_cores
+
+#detects the module names and returns the package
+def detect_md_engine(modulename):
+    """Detects the MD engine based on the available modules.
+        Any newly implemeted mdengines must be added here.
+    """
+    if 'gromacs' in modulename:
+        print(modulename)
+        return mdengines.gromacs
+    elif 'namd' in modulename:
+        return mdengines.namd
+    else:
+        raise RuntimeError("No Module Detected! did you specify the module?")
+
+def formatted_md_engine_name(modulename):
+    if 'gromacs' in modulename:
+        return "gromacs"
+    elif 'namd' in modulename:
+        return "namd"
+    else:
+        raise RuntimeError("No Module Detected! did you specify the module?")
```

Comment:
To complicated. 

In gromacs.py / namd.py
```python
import ...

NAME = "GROMACS"

def ...
```

Any function that later wants to name can do.

```python
get_md_engine(module).NAME
```

The idea is that every module carries a formatted name. This means we don't need to distinguish between the case where we want a name or a module.

Reply:
formatted_md_engine_name is not called anymore so it can be removed. as soon as the module was detected the respective engine files "know" their formatted and they can be hard coded.